lib/route.coffee

```diff
@@ -159,11 +159,12 @@ class @Route
     auth = @api.config.auth.user.call(endpointContext)
 
     # Get the user from the database
-    if not auth?.user and auth?.userId and auth?.token
-      userSelector = {}
-      userSelector._id = auth.userId
-      userSelector[@api.config.auth.token] = auth.token
-      auth.user = Meteor.users.findOne userSelector
+    if not auth?.user
+      if auth?.userId and auth?.token
```

Comment:
That's certainly a fair argument. For me, the extra layer of nesting makes me look for some further meaning, which slows me down a tiny bit. I wonder if a good compromise would be to rephrase it as 
```coffeescript
if auth?.userId and auth?.token and not auth?.user
```

What do you think of that? We get to keep the one-liner, and the ambiguity is removed. 

In reality, the `not` clause here isn't totally necessary. It only catches the weird situation where someone returned a `userId`, `token`, and `user` from their custom auth function (and skips the extra layer of auth if the `user` is included), when in reality they should only be returning either a `user` by itself or a `userId` and `token` together. After further review, I don't think the `userId` should be required either - but that will need to be taken care of in a separate PR.